--- langselector/locale.go.orig	2020-09-18 10:39:04.213018350 +0800
+++ langselector/locale.go	2020-09-18 10:43:56.901011981 +0800
@@ -33,15 +33,15 @@
 
 	"github.com/godbus/dbus"
 	// dbus services:
-	"github.com/linuxdeepin/go-dbus-factory/com.deepin.api.localehelper"
+	localehelper "github.com/linuxdeepin/go-dbus-factory/com.deepin.api.localehelper"
 	libnetwork "github.com/linuxdeepin/go-dbus-factory/com.deepin.daemon.network"
-	"github.com/linuxdeepin/go-dbus-factory/com.deepin.lastore"
-	"github.com/linuxdeepin/go-dbus-factory/org.freedesktop.notifications"
+	lastore "github.com/linuxdeepin/go-dbus-factory/com.deepin.lastore"
+	notifications "github.com/linuxdeepin/go-dbus-factory/org.freedesktop.notifications"
 	"pkg.deepin.io/dde/api/lang_info"
 	"pkg.deepin.io/dde/api/language_support"
 	"pkg.deepin.io/dde/api/userenv"
 	ddbus "pkg.deepin.io/dde/daemon/dbus"
-	"pkg.deepin.io/gir/gio-2.0"
+	gio "pkg.deepin.io/gir/gio-2.0"
 	"pkg.deepin.io/lib/dbusutil"
 	. "pkg.deepin.io/lib/gettext"
 	"pkg.deepin.io/lib/gsettings"
@@ -94,8 +94,8 @@
 var (
 	//save old language notifycation data
 	notifyTxtStartWithInstall string
-	notifyTxtStart string
-	notifyTxtDone string
+	notifyTxtStart            string
+	notifyTxtDone             string
 )
 
 //go:generate dbusutil-gen -type LangSelector locale.go
@@ -415,29 +415,7 @@
 	lang.setPropCurrentLocale(locale)
 	lang.PropsMu.Unlock()
 
-	// send notification
-	networkEnabled, err := isNetworkEnable()
-	if err != nil {
-		logger.Warning(err)
-	}
-
-	if networkEnabled {
-		sendNotify(localeIconStart, "", notifyTxtStartWithInstall)
-	} else {
-		sendNotify(localeIconStart, "", notifyTxtStart)
-	}
-
-	// generate locale
-	err = lang.generateLocale(locale)
-	if err != nil {
-		logger.Warning("failed to generate locale:", err)
-		lang.setLocaleFailed(oldLocale)
-		return
-	} else {
-		logger.Debug("generate locale success")
-	}
-
-	err = writeUserLocale(locale)
+	err := writeUserLocale(locale)
 	if err != nil {
 		logger.Warning("failed to write user locale:", err)
 		lang.setLocaleFailed(oldLocale)
@@ -461,16 +439,6 @@
 		}
 	}
 
-	// install language support packages
-	if networkEnabled {
-		err = lang.installLangSupportPackages(locale)
-		if err != nil {
-			logger.Warning("failed to install packages:", err)
-		} else {
-			logger.Debug("install packages success")
-		}
-	}
-
 	sendNotify(localeIconFinished, "", notifyTxtDone)
 
 	// end
